internal/service/dataexchange/event_action_test.go

```diff
@@ -0,0 +1,671 @@
+					acctest.MatchResourceAttrRegionalARN(ctx, resourceName, names.AttrARN, "dataexchange", regexache.MustCompile(`event-actions/.+`)),
+				),
+			},
+		},
+	})
+}
+
+func testAccEventActionPreCheck(ctx context.Context, t *testing.T) {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).DataExchangeClient(ctx)
+
+	input := &dataexchange.ListEventActionsInput{}
+
+	_, err := conn.ListEventActions(ctx, input)
+
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+
+func testAccCheckEventActionDestroy(ctx context.Context) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acctest.Provider.Meta().(*conns.AWSClient).DataExchangeClient(ctx)
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_dataexchange_event_action" {
+				continue
+			}
+
+			_, err := conn.GetEventAction(ctx, &dataexchange.GetEventActionInput{
+				EventActionId: aws.String(rs.Primary.ID),
+			})
+
+			if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+				return nil
+			}
+
+			if err != nil {
+				return create.Error(names.DataExchange, create.ErrActionCheckingDestroyed, tfdataexchange.ResNameEventAction, rs.Primary.ID, err)
+			}
+
+			return create.Error(names.DataExchange, create.ErrActionCheckingDestroyed, tfdataexchange.ResNameEventAction, rs.Primary.ID, errors.New("not destroyed"))
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckEventActionExists(ctx context.Context, n string, v *dataexchange.GetEventActionOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).DataExchangeClient(ctx)
+		output, err := conn.GetEventAction(ctx, &dataexchange.GetEventActionInput{
+			EventActionId: aws.String(rs.Primary.ID),
+		})
+		if err != nil {
+			return err
+		}
+		if output == nil {
+			return fmt.Errorf("DataExchange EventAction not found")
+		}
+
+		*v = *output
+
+		return nil
+	}
+}
+
+func testAccEventActionConfig_basic(bucketName, dataSetId, accountId string) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket" "test" {
+  bucket = "%s"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_policy" "test" {
+  bucket = aws_s3_bucket.test.id
+  policy = data.aws_iam_policy_document.test.json
+}
+
+data "aws_iam_policy_document" "test" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["dataexchange.amazonaws.com"]
+    }
+
+    actions = [
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.test.arn}/*",
+    ]
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+
+      values = [
+        "%s"
+      ]
+    }
+  }
+}
```

Comment:
Configuration shared across tests can be moved into its own helper function (e.g. `testAccEventActionConfigBase`) and then merged with test-specific config via the `acctest.ConfigCompose` helper function.